processor/spanmetricsprocessor/processor.go

```diff
@@ -225,9 +229,17 @@ func (p *processorImp) ConsumeTraces(ctx context.Context, traces ptrace.Traces)
 func (p *processorImp) tracesToMetrics(ctx context.Context, traces ptrace.Traces) error {
 	p.lock.Lock()
 
+	// Make sure metricKeys are empty in current batch
+	if len(p.metricKeys) != 0 {
+		p.metricKeys = []metricKey{}
+	}
```

Comment:
I don't think there's a need to conditionally reset `p.metricsKeys`; I thought we should always reset?
```suggestion
	// Make sure metricKeys are empty in current batch.
	p.metricKeys = []metricKey{}
```